addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -571,11 +572,74 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
     tagName: 'span',
     supportedFieldTypes: ['char'],
     isQuickEditable: true,
+    dynamicPlaceholderOptions: {
+        readonly: false,
+        needDefaultValue: true,
+        cancelOnEscape: true,
+        chainedTitle: true,
+        filter: (model) => !["one2many", "boolean", "many2many"].includes(model.type)
+    },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super(...arguments);
+        if (this.nodeOptions && this.nodeOptions.dynamic_placeholder) {
+            // When the dynamic placeholder is active,
+            // the recordData need to be updated when `mailing_model_real` change
+            this.resetOnAnyFieldChange = true;
+        }
+    },
+
+    openDynamicPlaceholder: async function (baseModel, chain = []) {
+        if (!baseModel) {
+            return;
+        }
```

Comment:
Could we please just _not_ call this function when there is no `baseModel` instead ?